zebra-network/src/peer_set/initialize.rs

```diff
@@ -896,7 +896,7 @@ where
                             // There weren't any peers, so try to get more peers.
```

Comment:
This PR doesn't fix these underlying bugs in the DemandHandshakeOrCrawl, or the tasks that trigger it:
- the syncer is not logging syncs or timeouts, it seems to be hung
- the mempool crawler isn't logging. Is it hung, has it stopped because we've fallen behind the tip (I didn't check the logs in the ticket for this), or do we need to add logs so we can see it being active?

If there is demand from the mempool crawler, then there aren't any valid peers, but there should be. So that could be another bug.

Reply:
To be clear, we don't need to fix everything in this PR, just enough to resolve the disconnection bug. But let's open tickets for the rest, so we don't forget.